x-pack/solutions/observability/plugins/apm_data_access/server/plugin.ts

```diff
@@ -10,63 +10,47 @@ import type {
   CoreSetup,
   CoreStart,
   Plugin,
-  SavedObjectsClientContract,
   Logger,
 } from '@kbn/core/server';
-import type { APMDataAccessConfig } from '.';
+import type { APMIndices } from '@kbn/apm-sources-access-plugin/server';
 import type {
   ApmDataAccessPluginSetup,
   ApmDataAccessPluginStart,
   ApmDataAccessServerDependencies,
+  ApmDataAccessServerSetupDependencies,
 } from './types';
-import { migrateLegacyAPMIndicesToSpaceAware } from './saved_objects/migrations/migrate_legacy_apm_indices_to_space_aware';
-import {
-  apmIndicesSavedObjectDefinition,
-  getApmIndicesSavedObject,
-} from './saved_objects/apm_indices';
 import { getServices } from './services/get_services';
 import type { ApmDataAccessPrivilegesCheck } from './lib/check_privileges';
 import { checkPrivileges } from './lib/check_privileges';
 
 export class ApmDataAccessPlugin
   implements Plugin<ApmDataAccessPluginSetup, ApmDataAccessPluginStart>
 {
-  public server?: ApmDataAccessServerDependencies;
-  public config: APMDataAccessConfig;
+  public config?: APMIndices;
   public logger: Logger;
 
   constructor(initContext: PluginInitializerContext) {
-    this.config = initContext.config.get<APMDataAccessConfig>();
     this.logger = initContext.logger.get();
   }
 
-  getApmIndices = async (savedObjectsClient: SavedObjectsClientContract) => {
-    const apmIndicesFromSavedObject = await getApmIndicesSavedObject(savedObjectsClient);
-    return { ...this.config.indices, ...apmIndicesFromSavedObject };
-  };
-
-  public setup(core: CoreSetup): ApmDataAccessPluginSetup {
-    // register saved object
-    core.savedObjects.registerType(apmIndicesSavedObjectDefinition);
+  public setup(
+    core: CoreSetup,
+    plugins: ApmDataAccessServerSetupDependencies
+  ): ApmDataAccessPluginSetup {
+    this.config = plugins.apmSourcesAccess.apmIndicesFromConfigFile;
 
     // expose
     return {
-      apmIndicesFromConfigFile: this.config.indices,
-      getApmIndices: this.getApmIndices,
+      apmIndicesFromConfigFile: this.config,
+      getApmIndices: plugins.apmSourcesAccess.getApmIndices,
```

Comment:
Do we still need to expose `getApmIndices` here? Wouldn't it be better to use it directly from `apm_sources_access` plugin?

Reply:
same for `apmIndicesFromConfigFile`

Reply:
This is more to reduce the scope of the changes to be more sane. I could add a TODO on this part for a follow-up PR, so I don't balloon the changes further.

Reply:
Let's do that, please.